src/kernel/arch/x86/pit.zig

```diff
@@ -148,111 +185,135 @@ inline fn sendCommand(cmd: u8) void {
 ///     IN data: u8              - The data to send.
 ///
 inline fn sendDateToCounter(comptime counter: u16, data: u8) void {
-    assert(counter == COUNTER_0_REGISTER or counter == COUNTER_1_REGISTER or counter == COUNTER_2_REGISTER);
+    comptime expect(counter == COUNTER_0_REGISTER or counter == COUNTER_1_REGISTER or counter == COUNTER_2_REGISTER);
     arch.outb(counter, data);
 }
 
-fn pitHandler(context: *arch.InterruptContext) void {
-    ticks += 1;
+///
+/// The interrupt handler for the PIT. This will increment a counter for now.
+///
+/// Arguments:
+///     IN ctx: *arch.InterruptContext - Pointer to the interrupt context containing the contents
+///                                      of the register at the time of the interrupt.
+///
+fn pitHandler(ctx: *arch.InterruptContext) void {
+    ticks +%= 1;
 }
 
 ///
 /// Set up a counter with a tick rate and mode of operation.
 ///
 /// Arguments:
-///     IN freq: u16   - The frequency that the counter operates at.
 ///     IN counter: u8 - Which counter is to be set up, either OCW_SELECT_COUNTER_0,
 ///                      OCW_SELECT_COUNTER_1 or OCW_SELECT_COUNTER_2 only.
+///     IN freq: u32   - The frequency that the counter operates at. Any frequency that is between
```

Comment:
I would suggest throwing an error or doing a comptime check on the frequency instead of setting to a min/max value if it's outside the range